### PR TITLE
feat: metadata crawler

### PR DESCRIPTION
**Pull request by yassilah**
Refers to: https://github.com/nuxt-modules/algolia/issues/8

### CODE REVIEW

File: src/crawler/index.ts [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+/**
+ * Default metadata getter using "metascaper".
+ */
+function createDefaultMetaGetter () {
+  const scraper = metascraper([])
+
+  return async (html: string, route: string) => {
+    return await scraper({ html, url: route })
+  }
+}
```
Comment (Baroshem) [inline src/crawler/index.ts:50]: What do you think about extracting these helper functions into src/utils/index.ts? These functions are quite useful but for someone who would like to focus on the main purpose of this file they are not that important. Buy moving them to utils we are extracting the logic to other files which makes it easier to analyze :)

File: src/crawler/index.ts [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+      await nuxt.callHook('crawler:index:after', {
+        options,
+        pages,
+        client,
+        index
+      })
+    }
+  }
+}
```
Comment (Baroshem) [inline src/crawler/index.ts:112]: For these hooks, I have been thinking about creating a src/hooks/crawler.ts file. What do you think?

File: src/module.ts [hunk 3]
```diff
@@ -16,6 +17,23 @@ export default defineNuxtModule<AlgoliaOptions>({
       throw new Error('Missing `applicationId`')
     }
 
+    if (options.crawler) {
```
Comment (Baroshem) [inline src/module.ts:20]: What is missing here are the docs explaining how to use the crawler. Could you please add a information in this page about the new options and the types similarly to how it was done before? https://github.com/nuxt-modules/algolia/blob/main/docs/pages/2.setup.md

File: src/module.ts [hunk 4]
```diff
@@ -16,6 +17,23 @@ export default defineNuxtModule<AlgoliaOptions>({
...
+        throw new Error('Missing `crawler.apiAdminKey`')
+      }
+
+      if (!options.crawler.indexName) {
+        throw new Error('Missing `crawler.indexName`')
+      }
+
+      const pages: CrawlerPage[] = []
+
+      nuxt.addHooks({
```
Comment (Baroshem) [inline src/module.ts:31]: Also, you could create a new page here called 4.advanced.md with the explanation of how crawler works and how the user can utilize it to extend their needs -> https://github.com/nuxt-modules/algolia/tree/main/docs/pages

**Review by Baroshem — COMMENTED** [review]
And I believe that will be all of my CR recommendations. 

Really a lot of amazing work! No code changes, just recommendations regarding file structure and the docs. Cant wait to merge it and release it (also mentioning that amazing contribution!)

File: src/crawler/index.ts (hunk 1 above)
Comment (yassilah) [reply src/crawler/index.ts:50]: Hey :) Sure, I can do that but moving it into a utils folder make it seem like they could be reused for some other purpose and in this case I don't really know what they could be used for besides that specific feature?

File: src/crawler/index.ts (hunk 2 above)
Comment (yassilah) [reply src/crawler/index.ts:112]: Yes that's a good idea to gather hooks together

**Pull request merged**